Панин Максим

PaninMaxim opened the pull request.

PaninMaxim committed:
Add files via upload

PaninMaxim committed:
Rename two_sum.cpp to two-sum.cpp

PaninMaxim committed:
Update two-sum.cpp

czertyaka commented on src/two-sum.cpp:
```diff
@@ -0,0 +1,39 @@
+#include <iostream>
+#define ARRAY_SIZE 5
```
Можно подключить заголовочный файл two-sum.hpp и не дублировать определение макроса

czertyaka commented on src/two-sum.cpp:
```diff
@@ -0,0 +1,39 @@
+#include <iostream>
```
Заголовочный файл не используется

czertyaka commented on src/two-sum.cpp:
```diff
@@ -0,0 +1,39 @@
+	std::size_t& index1
+)
+{
+	unsigned int kolvo_par = 0;
```
Зачем считаете количество пар? По условию задачи подойдет любая пара

czertyaka commented on src/two-sum.cpp:
```diff
@@ -0,0 +1,39 @@
+	bool proverka_par = false;
+	for (unsigned int i = 0; i <= ARRAY_SIZE - 1; i++)
+	{
+		for (unsigned int j = i + 1; j <= ARRAY_SIZE; j++)
```
Условие должно быть `< ARRAY_SIZE`. Сейчас имеем выход за границы массива, что показывают [тесты](https://github.com/czertyaka/CppDevCourse-hw3/actions/runs/12107085066/job/34090372814?pr=13)

PaninMaxim committed:
Update two-sum.cpp

czertyaka commented on src/two-sum.cpp:
```diff
@@ -0,0 +1,22 @@
+{
+	for (unsigned int i = 0; i < ARRAY_SIZE; i++)
+	{
+		for (unsigned int j = i + 1; j <= ARRAY_SIZE; j++)
```
В этой строке все еще условие неверно

PaninMaxim replied:
Здравствуйте, при i = 0, j = 1...5
при i = 1, j = 2...5
при i = 2, j = 3 4 5
при i = 3, j = 4 5
при i = 4, j = 5
i не будет равно 5 никогда, т.к. знак меньше
j не превысит 5, т.к. знак меньше-равно

PaninMaxim committed:
Update two-sum.cpp

Если я всё правильно понял, то ошибка в том что я забыл что счёт идёт с 0, и до 5(ARRAY_SIZE) будет всего 6 чисел, а не 5, как должно быть

czertyaka commented:
Задание принято
